market_maker_keeper/leverj_market_maker_keeper.py

```diff
@@ -0,0 +1,244 @@
+        self.control_feed = create_control_feed(self.arguments)
+        self.order_history_reporter = create_order_history_reporter(self.arguments)
+
+        self.history = History()
+
+        self.leverj_api = LeverjAPI(web3=self.web3,
+                                    api_server=self.arguments.leverj_api_server,
+                                    account_id=self.arguments.account_id,
+                                    api_key=self.arguments.api_key,
+                                    api_secret=self.arguments.api_secret,
+                                    timeout=self.arguments.leverj_timeout)
+
+
+        self.order_book_manager = OrderBookManager(refresh_frequency=self.arguments.refresh_frequency)
+        self.order_book_manager.get_orders_with(lambda: self.leverj_api.get_orders(self.pair()))
+        self.order_book_manager.get_balances_with(lambda: self.leverj_api.get_balances())
+        self.order_book_manager.cancel_orders_with(lambda order: self.leverj_api.cancel_order(order.order_id))
+        self.order_book_manager.enable_history_reporting(self.order_history_reporter, self.our_buy_orders,
+                                                         self.our_sell_orders)
+        self.order_book_manager.start()
+
+    def main(self):
+        with Lifecycle() as lifecycle:
+            lifecycle.initial_delay(1)
+            lifecycle.on_startup(self.startup)
+            lifecycle.every(1, self.synchronize_orders)
+            lifecycle.on_shutdown(self.shutdown)
+
+    def startup(self):
+        quote_increment = 1/(self.leverj_api.get_product(self.arguments.pair)["ticksperpoint"])
+        self.precision = -(int(log10(float(quote_increment)))+1)
+
+    def shutdown(self):
+        self.order_book_manager.cancel_all_orders()
+
+    def pair(self):
+        return self.arguments.pair.upper()
+
+    def token_sell(self) -> str:
+        if self.arguments.pair.startswith("USDC"):
+            return "USDC"
+        return self.arguments.pair[:3]
+
+    def token_buy(self) -> str:
+        if self.arguments.pair.startswith("USDC"):
+            return self.arguments.pair[4:]
+        return self.arguments.pair[3:]
+
+
+    def our_available_balance(self, our_balances: dict, token: str) -> Wad:
+        for key in our_balances:
+            if our_balances[key]['symbol'] == token:
+                if (token == "LEV") or (token == "FEE"): 
+                    return Wad(int(our_balances[key]['available'])*10**9)
+                elif (token == "USDC") or (token == "USDT"):
+                    return Wad(int(our_balances[key]['available'])*10**12)
+                elif (token == "WBTC"):
+                    return Wad(int(our_balances[key]['available'])*10**10)
+                elif (token == "GUSD"):
+                    return Wad(int(our_balances[key]['available'])*10**16)
+                else:
+                    return Wad(int(our_balances[key]['available']))
```

Comment:
Wouldn't the default case require multiplying by `10**18`, or using `Wad.from_number(...)`?

Reply:
No, for normal tokens the exchange passes back strings that in the base number, so for normal tokens like DAI, it's already in 10**18 format but just in a string case.